source/Halibut/Transport/WebSocketConnectionFactory.cs

```diff
@@ -58,8 +63,11 @@ ClientWebSocket CreateConnectedClient(ServiceEndPoint serviceEndpoint)
             {
                 ServerCertificateInterceptor.Expect(connectionId);
                 using (var cts = new CancellationTokenSource(serviceEndpoint.TcpClientConnectTimeout))
-                    client.ConnectAsync(serviceEndpoint.BaseUri, cts.Token)
-                        .ConfigureAwait(false).GetAwaiter().GetResult();
+                {
+                    using (cancellationToken.Register(() =>  cts?.Cancel()))
```

Comment:
Connect up the cancellation token so that we stop waiting if the caller triggers the cancel.